datasize/src/lib.rs

```diff
@@ -349,6 +349,25 @@ where
     }
 }
 
+impl<T, E> DataSize for Result<T, E>
+where
+    T: DataSize,
+    E: DataSize,
+{
+    // Results are only not dynamic if their types have no heap data at all and are not dynamic.
+    const IS_DYNAMIC: bool =
+        (T::IS_DYNAMIC || E::IS_DYNAMIC || T::STATIC_HEAP_SIZE > 0 || E::STATIC_HEAP_SIZE > 0);
```

Comment:
I believe it's actually `|| (T::STATIC_HEAP_SIZE != E::STATIC_HEAP_SIZE)`. As long as the static heap size is the same for both variants, nothing changes. Otherwise it does.

Reply:
Ahh, I get it now. Yes, you're probably right :+1: